drivers/include/periph/spi.h

```diff
@@ -1,268 +1,299 @@
+ * in power down state.
  *
- * In slave mode the SPI device is purely reacting to the bus. Transaction will be started and
- * ended by a connected SPI master. When a byte is received, the callback is called in interrupt
- * context with this byte as argument. The return byte of the callback is transferred to the
- * master in the next transmission cycle. This interface enables easy implementation of a register
- * based access paradigm for the SPI slave.
+ * This function is intended to be called by the board initialization code
+ * during system startup to prepare the (shared) SPI device for further usage.
+ * It uses the board specific initialization parameters as defined in the
+ * board's `periph_conf.h`.
  *
- * @param[in] dev       The SPI device to initialize as SPI slave
- * @param[in] conf      Mode of clock phase and polarity
- * @param[in] cb        callback called every time a byte was received
+ * @param[in] bus       SPI device to initialize
  *
- * @return              0 on success
- * @return              -1 on error
+ * @return              SPI_OK on success
+ * @return              SPI_NODEV on invalid device
  */
-int spi_init_slave(spi_t dev, spi_conf_t conf, char (*cb)(char data));
+int spi_init(spi_t bus);
 
 /**
- * @brief Configure SCK, MISO and MOSI pins for the given SPI device
+ * @brief   Initialize the given chip select pin
  *
- * @param[in] dev       SPI device to use
+ * The chip select can be any generic GPIO pin (e.g. GPIO_PIN(x,y)), or it can
+ * be a hardware chip select line. The existence and number of of hardware chip
+ * select lines depends on the underlying platform and the actual pins used for
+ * hardware chip select lines are defined in the board's `periph_conf.h`.
  *
- * @return              0 on success
- * @return              -1 on error
+ * Define the used chip select line using the @ref SPI_HWCS(x) macro for
+ * hardware chip select line `x` or the GPIO_PIN(x,y) macro for using any
+ * GPIO pin for manual chip select.
+ *
+ * @return              SPI_OK on success
+ * @return              SPI_NODEV on invalid device
+ * @return              SPI_NOCS on invalid CS pin/line
  */
-int spi_conf_pins(spi_t dev);
+int spi_init_cs(spi_t bus, spi_cs_t cs);
 
 /**
- * @brief Get mutually exclusive access to the given SPI bus
- *
- * In case the SPI device is busy, this function will block until the bus is free again.
- *
- * @param[in] dev       SPI device to access
- *
- * @return              0 on success
- * @return              -1 on error
+ * @brief   Start a new SPI transaction
+ *
+ * Starting a new SPI transaction will get exclusive access to the SPI bus
+ * and configure it according to the given values. If another SPI transaction
+ * is active when this function is called, this function will block until the
+ * other transaction is complete (spi_relase was called).
+ *
+ * @param[in] bus       SPI device to access
+ * @param[in]  cs       chip select pin/line to use, may be GPIO_UNDEF
```

Comment:
Here and at the other occurances: Can you add a half-sentence what choosing `GPIO_UNDEF` means? I know it's documented in the overall detailed section above, but if you are just looking at the doc for this function (and the others), than you wouldn't be the wiser.